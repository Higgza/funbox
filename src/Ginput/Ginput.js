import React, {Component} from 'react';
import Geosuggest from 'react-geosuggest';
import './Ginput.css';

/* Redux store */
import {connect} from 'react-redux';
import {addAdress, addMapArray} from "../redux/actions/MapActions/mapActions";

export class Gimput extends Component {

    geoSuggest = (node) => {
        this._geoSuggest = node;
    };

    onAddAdressEnter = (e) => {
        if (e.which == null) {
            if (e.keyCode === 13) {
                this.props.onAddMapArray(this.props.mapInfo);
                this._geoSuggest.clear();
            }
        }

        if (e.which !== 0 && e.charCode !== 0) {
            if(e.which === 13) {
                this.props.onAddMapArray(this.props.mapInfo);
                this._geoSuggest.clear();
            }
        }

    };

    render() {
        return (
            <div>
                <Geosuggest ref={this.geoSuggest}
                            onKeyPress={this.onAddAdressEnter}
                            queryDelay={0}
                            ignoreTab={true}
                            placeholder="Укажите точку маршрута"
                            /*location={new window.google.maps.LatLng(55.159283, 61.381320)}*/
                            location={this.props.location}
                            radius={20} onSuggestSelect={this.props.onaddAdress} />
                <button className={'geobtn'}
                        disabled={this.props.mapInfo === undefined || this.props.mapInfo === ''}
                        onClick={() => {
                            this.props.onAddMapArray(this.props.mapInfo);
                            this._geoSuggest.clear();
                }}>Add for route</button>
            </div>
        );
    }
}

function s(state) {
    return {
        mapInfo: state.googleMap.info,
        /*mapArray: state.googleMap.arr*/
        location: new window.google.maps.LatLng(55.159283, 61.381320),
    }
}

function d(dispatch) {
    return {
        onaddAdress: (label) => dispatch(addAdress(label)),
        onAddMapArray: (label) => dispatch(addMapArray(label))
    }
}

export default connect(s, d)(Gimput);