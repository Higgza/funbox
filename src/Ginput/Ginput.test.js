import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Gimput} from './Ginput';

configure({
    adapter: new Adapter()
});

describe('Ginput component', () => {

    const props = {
        mapInfo: '',
        onaddAdress: jest.fn(),
        onAddMapArray: jest.fn(),
        location: jest.fn(),
    };

    describe('Initial Ginput component', () => {

        const initialProps = {
            ...props
        };

        const GComponent = shallow(<Gimput {...initialProps}  />);

        it('snapshot initial Ginput component', () => {
            expect(GComponent).toMatchSnapshot();
        });

        it('disabled button Gunput', () => {
            expect(GComponent.find('button').prop('disabled')).toBeTruthy();
        });
    });

    describe('Props Ginput component', () => {

        const nextProps = {
            ...props,
            mapInfo: 'string'
        };

        const GComponent = shallow(<Gimput {...nextProps}  />);

        it('snapshot Props Ginput component', () => {
            expect(GComponent).toMatchSnapshot();
        });

        it('button desabled test', () => {
            expect(GComponent.find('button').prop('disabled')).toBeFalsy();
        });

        /*
        it('button onClick test', () => {
            GComponent.find('button').simulate('click');
            expect(nextProps.onAddMapArray()).toHaveBeenCalled();
        });*/

    });

});