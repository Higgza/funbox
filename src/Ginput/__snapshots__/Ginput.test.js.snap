// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ginput component Initial Ginput component snapshot initial Ginput component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gimput
    location={[MockFunction]}
    mapInfo=""
    onAddMapArray={[MockFunction]}
    onaddAdress={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <default_1$$1
          autoActivateFirstSuggest={false}
          disabled={false}
          fixtures={Array []}
          getSuggestLabel={[Function]}
          highlightMatch={true}
          ignoreEnter={false}
          ignoreTab={true}
          initialValue=""
          location={[MockFunction]}
          maxFixtures={10}
          minLength={1}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onSuggestSelect={[MockFunction]}
          placeholder="Укажите точку маршрута"
          queryDelay={0}
          radius={20}
          skipSuggest={[Function]}
          style={Object {}}
        />,
        <button
          className="geobtn"
          disabled={true}
          onClick={[Function]}
        >
          Add for route
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoActivateFirstSuggest": false,
          "disabled": false,
          "fixtures": Array [],
          "getSuggestLabel": [Function],
          "highlightMatch": true,
          "ignoreEnter": false,
          "ignoreTab": true,
          "initialValue": "",
          "location": [MockFunction],
          "maxFixtures": 10,
          "minLength": 1,
          "onKeyDown": [Function],
          "onKeyPress": [Function],
          "onSuggestSelect": [MockFunction],
          "placeholder": "Укажите точку маршрута",
          "queryDelay": 0,
          "radius": 20,
          "skipSuggest": [Function],
          "style": Object {},
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add for route",
          "className": "geobtn",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add for route",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <default_1$$1
            autoActivateFirstSuggest={false}
            disabled={false}
            fixtures={Array []}
            getSuggestLabel={[Function]}
            highlightMatch={true}
            ignoreEnter={false}
            ignoreTab={true}
            initialValue=""
            location={[MockFunction]}
            maxFixtures={10}
            minLength={1}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onSuggestSelect={[MockFunction]}
            placeholder="Укажите точку маршрута"
            queryDelay={0}
            radius={20}
            skipSuggest={[Function]}
            style={Object {}}
          />,
          <button
            className="geobtn"
            disabled={true}
            onClick={[Function]}
          >
            Add for route
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoActivateFirstSuggest": false,
            "disabled": false,
            "fixtures": Array [],
            "getSuggestLabel": [Function],
            "highlightMatch": true,
            "ignoreEnter": false,
            "ignoreTab": true,
            "initialValue": "",
            "location": [MockFunction],
            "maxFixtures": 10,
            "minLength": 1,
            "onKeyDown": [Function],
            "onKeyPress": [Function],
            "onSuggestSelect": [MockFunction],
            "placeholder": "Укажите точку маршрута",
            "queryDelay": 0,
            "radius": 20,
            "skipSuggest": [Function],
            "style": Object {},
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add for route",
            "className": "geobtn",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add for route",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Ginput component Props Ginput component snapshot Props Ginput component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gimput
    location={[MockFunction]}
    mapInfo="string"
    onAddMapArray={[MockFunction]}
    onaddAdress={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <default_1$$1
          autoActivateFirstSuggest={false}
          disabled={false}
          fixtures={Array []}
          getSuggestLabel={[Function]}
          highlightMatch={true}
          ignoreEnter={false}
          ignoreTab={true}
          initialValue=""
          location={[MockFunction]}
          maxFixtures={10}
          minLength={1}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onSuggestSelect={[MockFunction]}
          placeholder="Укажите точку маршрута"
          queryDelay={0}
          radius={20}
          skipSuggest={[Function]}
          style={Object {}}
        />,
        <button
          className="geobtn"
          disabled={false}
          onClick={[Function]}
        >
          Add for route
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoActivateFirstSuggest": false,
          "disabled": false,
          "fixtures": Array [],
          "getSuggestLabel": [Function],
          "highlightMatch": true,
          "ignoreEnter": false,
          "ignoreTab": true,
          "initialValue": "",
          "location": [MockFunction],
          "maxFixtures": 10,
          "minLength": 1,
          "onKeyDown": [Function],
          "onKeyPress": [Function],
          "onSuggestSelect": [MockFunction],
          "placeholder": "Укажите точку маршрута",
          "queryDelay": 0,
          "radius": 20,
          "skipSuggest": [Function],
          "style": Object {},
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add for route",
          "className": "geobtn",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add for route",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <default_1$$1
            autoActivateFirstSuggest={false}
            disabled={false}
            fixtures={Array []}
            getSuggestLabel={[Function]}
            highlightMatch={true}
            ignoreEnter={false}
            ignoreTab={true}
            initialValue=""
            location={[MockFunction]}
            maxFixtures={10}
            minLength={1}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onSuggestSelect={[MockFunction]}
            placeholder="Укажите точку маршрута"
            queryDelay={0}
            radius={20}
            skipSuggest={[Function]}
            style={Object {}}
          />,
          <button
            className="geobtn"
            disabled={false}
            onClick={[Function]}
          >
            Add for route
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoActivateFirstSuggest": false,
            "disabled": false,
            "fixtures": Array [],
            "getSuggestLabel": [Function],
            "highlightMatch": true,
            "ignoreEnter": false,
            "ignoreTab": true,
            "initialValue": "",
            "location": [MockFunction],
            "maxFixtures": 10,
            "minLength": 1,
            "onKeyDown": [Function],
            "onKeyPress": [Function],
            "onSuggestSelect": [MockFunction],
            "placeholder": "Укажите точку маршрута",
            "queryDelay": 0,
            "radius": 20,
            "skipSuggest": [Function],
            "style": Object {},
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add for route",
            "className": "geobtn",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add for route",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
